source/common/network/connection_impl.cc

```diff
@@ -102,7 +102,7 @@ void ConnectionImpl::close(ConnectionCloseType type) {
     if (data_to_write > 0) {
       // We aren't going to wait to flush, but try to write as much as we can if there is pending
       // data.
-      transport_socket_->doWrite(*write_buffer_);
+      transport_socket_->doWrite(*write_buffer_, true);
```

Comment:
a quick (lazy) scan of the connection_impl_test.cc has me seeing mostly doWrite(_, _)
Can you make sure relevant tests here and elsewhere we hard-code (ssl test) see the expected boolean?